server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/VacationIntegrationTest.java

```diff
@@ -0,0 +1,212 @@
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.RestAssured.with;
+import static com.jayway.restassured.config.EncoderConfig.encoderConfig;
+import static com.jayway.restassured.config.RestAssuredConfig.newConfig;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.jmap.api.access.AccessToken;
+import org.apache.james.mailbox.model.MailboxConstants;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.base.Charsets;
+import com.jayway.awaitility.Awaitility;
+import com.jayway.awaitility.Duration;
+import com.jayway.awaitility.core.ConditionFactory;
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.http.ContentType;
+
+public abstract class VacationIntegrationTest {
+
+    private static final String ARGUMENTS = "[0][1]";
+    private static final String SECOND_NAME = "[1][0]";
+    private static final String SECOND_ARGUMENTS = "[1][1]";
+    private static final String DOMAIN = "mydomain.tld";
+    private static final String USER_1 = "benwa@" + DOMAIN;
+    private static final String USER_2 = "matthieu@" + DOMAIN;
+    private static final String PASSWORD = "secret";
+    private static final String REASON = "Message explaining my wonderful vacations";
+
+    private ConditionFactory calmlyAwait;
+
+    protected abstract GuiceJamesServer<?> createJmapServer();
+
+    protected abstract void await();
+
+    @Before
+    public void setUp() throws Exception {
+        GuiceJamesServer<?> guiceJamesServer = createJmapServer();
+        guiceJamesServer.start();
+
+        guiceJamesServer.serverProbe().addDomain(DOMAIN);
+        guiceJamesServer.serverProbe().addUser(USER_1, PASSWORD);
+        guiceJamesServer.serverProbe().addUser(USER_2, PASSWORD);
+        guiceJamesServer.serverProbe().createMailbox(MailboxConstants.USER_NAMESPACE, USER_2, "outbox");
+        guiceJamesServer.serverProbe().createMailbox(MailboxConstants.USER_NAMESPACE, USER_1, "sent");
+        guiceJamesServer.serverProbe().createMailbox(MailboxConstants.USER_NAMESPACE, USER_2, "sent");
+        await();
+
+
+        RestAssured.port = guiceJamesServer.getJmapPort();
+        RestAssured.config = newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8));
+
+        Duration slowPacedPollInterval = Duration.FIVE_HUNDRED_MILLISECONDS;
+        calmlyAwait = Awaitility.with().pollInterval(slowPacedPollInterval).and().with().pollDelay(slowPacedPollInterval).await();
+    }
+
+    @Test
+    public void jmapVacationShouldGenerateAReplyWhenActive() throws Exception {
```

Comment:
You could add other integration tests: when inactive, when dates don't match, etc.
To reduce copy/paste and enhance readability, you can use cucumber for these tests.